File: backend/pkg/model/section.go
package model

type Section struct {
	SectionId int    `json:"section_id" db:"id"`
	Name      string `json:"name" db:"name"`
}

type AddSectionInput struct {
	Name         string `json:"name" binding:"required"`
	NameEn       string `json:"name_en" binding:"required"`
	DisciplineId int    `json:"discipline_id" binding:"required"`
}

type DeleteSectionInput struct {
	Id int `json:"section_id" binding:"required"`
}

type ChangeSectionNameInput struct {
	Id   int    `json:"section_id" binding:"required"`
	Name string `json:"name" binding:"required"`
}

type GetDisciplineSectionsInput struct {
	DisciplineId int `json:"discipline_id" binding:"required"`
}

type SectionsResponse struct {
	Ru []Section `json:"ru"`
	En []Section `json:"en"`
}

type TestToSectionInput struct {
	SectionId int `json:"section_id" binding:"required"`
	TestId    int `json:"test_id" binding:"required"`
}

type LabToSectionInput struct {
	SectionId     int `json:"section_id" binding:"required"`
	ExternalLabId int `json:"external_lab_id" binding:"required"`
	DefaultMark   int `json:"default_mark" binding:"required"`
}

File: frontend/src/models/section/section.ts
export type ISection = ISection.Section;

export namespace ISection {
  export interface Section {
    name: string;
    section_id: number;
  }

  export interface GetSections {
    en: ISection[];
    ru: ISection[];
  }

  export interface CreateSection {
    name: string;
    name_en: string;
    discipline_id: number;
  }

  export interface AddLabToSection {
    external_lab_id: number;
    section_id: number;
    default_mark: number;
  }

  export interface AddTestToSection {
    test_id: number;
    section_id: number;
  }

  export interface DeleteLabFromSection {
    laboratory_id: number;
    section_id: number;
  }
}
